pool/tick_bitmap.gno

```diff
@@ -51,63 +47,91 @@ func (pool *Pool) tickBitmapNextInitializedTickWithInOneWord(
-		mask = new(u256.Uint).Sub(mask, u256.NewUint(1))
-		mask = new(u256.Uint).Add(mask, shift1LeftByBitPos.Clone())
+// generateFullMaskForBitPos generates a mask with all bits set to 1 up to the given bit position
+func generateFullMaskForBitPos(bitPos uint) *u256.Uint {
+	// (1 << bitPos) - 1 + (1 << bitPos)
+	shifted := new(u256.Uint).Lsh(u256.One(), bitPos)
+	minusOne := new(u256.Uint).Sub(shifted, u256.One())
 
-		pool.tickBitmaps[wordPos] = pool.tickBitmaps[wordPos].NilToZero()
-		masked := new(u256.Uint).And(pool.tickBitmaps[wordPos], mask)
+	return new(u256.Uint).Add(minusOne, shifted)
+}
 
-		initialized := !(masked.IsZero())
+// generateMaskClearingBitsBelow generates a mask with all bits set to 1 below the given bit position
+func generateMaskClearingBitsBelow(bitPos uint) *u256.Uint {
+	// ~((1 << bitPos) - 1)
+	shifted := new(u256.Uint).Lsh(u256.One(), bitPos)
+	minusOne := new(u256.Uint).Sub(shifted, u256.One())
 
-		if initialized {
-			next := (compress - int32(bitPos-plp.BitMathMostSignificantBit(masked))) * tickSpacing
-			return next, initialized
-		}
+	return new(u256.Uint).Not(minusOne)
+}
 
-		next := (compress - int32(bitPos)) * tickSpacing
-		return next, initialized
+// getTickBitmap gets the tick bitmap for the given word position
+// if the tick bitmap is not initialized, initialize it to zero
+func (pool *Pool) getTickBitmap(wordPos int16) *u256.Uint {
+	if pool.tickBitmaps[wordPos] == nil {
+		pool.tickBitmaps[wordPos] = u256.Zero()
 	}
 
-	wordPos, bitPos := tickBitmapPosition(compress + 1)
+	return pool.tickBitmaps[wordPos]
+}
 
-	shift1LeftByBitPos := new(u256.Uint).Lsh(u256.NewUint(1), uint(bitPos))
-	_mask := shift1LeftByBitPos.Clone()
-	_mask = new(u256.Uint).Sub(_mask, u256.NewUint(1))
-	mask := bitwiseNotForUint256BitmapIndexing(_mask)
+// setTickBitmap sets the tick bitmap for the given word position
+func (pool *Pool) setTickBitmap(wordPos int16, bitmap *u256.Uint) {
+	pool.tickBitmaps[wordPos] = bitmap
+}
 
-	pool.tickBitmaps[wordPos] = pool.tickBitmaps[wordPos].NilToZero()
-	masked := new(u256.Uint).And(pool.tickBitmaps[wordPos], mask)
+// getWordAndBitPos gets tick's wordPos and bitPos depending on the swap direction
+func getWordAndBitPos(tick int32, lte bool) (int16, uint8) {
+	if lte {
+		return tickBitmapPosition(tick)
+	}
 
-	initialized := !(masked.IsZero())
-	if initialized {
-		next := (compress + 1 + int32(plp.BitMathLeastSignificantBit(masked)-bitPos)) * tickSpacing
-		return next, initialized
-	} else {
-		next := (compress + 1 + int32(255-bitPos)) * tickSpacing
+	tick++
+	return tickBitmapPosition(tick)
+}
 
-		return next, initialized
+// getMaskBit gets a mask with all bits set to 1 up to the given bit position if lte is true,
+// or a mask with all bits set to 1 below the given bit position if lte is false
+func getMaskBit(bitPos uint, lte bool) *u256.Uint {
+	if lte {
+		return generateFullMaskForBitPos(bitPos)
+	}
+
+	return generateMaskClearingBitsBelow(bitPos)
+}
```

Comment:
```suggestion
var bmap = map[bool]uint{
	true:  1,
	false: 0,
}

// getMaskBit gets a mask with all bits set to 1 up to the given bit position if lte is true,
// or a mask with all bits set to 1 below the given bit position if lte is false
func getMaskBit(bitPos uint, lte bool) *u256.Uint {
	shifted := new(u256.Uint).Lsh(u256.One(), bitPos+bmap[lte])
	mask := new(u256.Uint).Sub(shifted, u256.One())

	if !lte {
		return new(u256.Uint).Not(mask)
	}
	return mask
}
```

Split function names are understood differently than they are intended.
We modified the proposed code to compute in unison.